src/contracts/core/PermissionController.sol

```diff
@@ -0,0 +1,117 @@
+     * by message senders that are admins for the given account.
+     *
+     */
+    modifier onlyAccountAdmin(address account) {
+        require(accounts[account].admins.contains(msg.sender), CallerNotAdmin());
+        _;
+    }
+
+    //////////////////////////////////////////////////
+    // Public Management Interface
+    //////////////////////////////////////////////////
+    
+    /// @inheritdoc IPermissionController
+    function createAccount() external {
+        // we assume the message sender is an AVS, Operator, or Staker, but we don't
+        // necessarily verify this. There is no harm in any address creating its own
+        // delegated account. It's not meaningfully possible to "grief" account creation.
+        DelegatedAccountInfo storage account = accounts[msg.sender];
+
+        // check to make sure the account doesn't already exist
+        require(!account.isValid, AccountAlreadyExists());
+
+        // make the account valid by marking it as such, and by default add the
+        // message sender as an admin. The account address is not a permanent
+        // admin, so its inclusion in the set is critical.
+        account.isValid = true;
+        account.admins.add(msg.sender);
+
+        emit DelegatedAccountCreated(msg.sender);
+    }
+
+    /// @inheritdoc IPermissionController
+    function setAccountAdmin(address account, address delegate, bool isAdmin) onlyValidAccounts(account) onlyAccountAdmin(account) external {
+        // at this point we know the caller is admin on a valid account, so
+        // we can simply set the state and emit an event
+        if (isAdmin) {
+            accounts[account].admins.add(delegate);
+        } else {
+            accounts[account].admins.remove(delegate);
+        }
+
+        emit DelegatedAccountAdminChange(account, msg.sender, delegate, isAdmin);
+    }
+
+    /// @inheritdoc IPermissionController
+    function setDelegatedRole(
+        address account,
+        address target,
+        bytes4 selector,
+        address delegate,
+        bool hasPermission) onlyValidAccounts(account) onlyAccountAdmin(account) external {
+        
+        // at this point we know the caller is admin on a valid account,
+        // so we can simply set the state, update the index and emit an event
+        DelegatedAccountInfo storage info = accounts[account];
+        info.delegations[delegate][target][selector] = hasPermission;
+        if (hasPermission) {
+            info.delegatedContractSelectors[delegate][target].add(selector);
+        } else {
+            info.delegatedContractSelectors[delegate][target].remove(selector);
+        }
+
+        emit AccountDelegateStateChange(account, msg.sender, target, selector, delegate, hasPermission); 
+    }
+    
+    //////////////////////////////////////////////////
+    // Public Introspection Interface
+    //////////////////////////////////////////////////
```

Comment:
Is the idea that the core protocol functions would check one of these view functions to see if it's an admin or if it's a delegate and has the right permissions?